Scrape proptery details

gbp opened the pull request:
Add ability to scrape property pages by calling the `include_property` method, see below:

```ruby
UKPlanningScraper::Authority.named('Cardiff').include_property.decided_days(4).scrape
```

This will add the address details, UPRN, ward and parish information to the applications.

gbp committed:
Don't scrape properties by default

gbp committed:
Extract summary scraping into separate method

gbp committed:
Scape properties details

gbp committed:
Extract common Mechanize error handling

chrismytton commented on lib/uk_planning_scraper/idox.rb:
```diff
@@ -186,9 +188,73 @@ def parse_info_url(app)
       end # if
     end
+
+    def parse_proptery_url(app)
```
Typo in method name: should be `parse_property_url`

chrismytton commented on lib/uk_planning_scraper/idox.rb:
```diff
@@ -186,9 +188,73 @@ def parse_info_url(app)
+      end
+    end
+
+    def parse_proptery_detail_urls(app)
```
Typo in method name: should be `parse_property_detail_urls`

chrismytton commented on lib/uk_planning_scraper/idox.rb:
```diff
@@ -186,9 +188,73 @@ def parse_info_url(app)
         end # each row
+
+        # find associated property link
+        property_assoication_link = res.at('p.associatedproperty a')
```
Typo in variable name: should be `property_association_link`

chrismytton reviewed (commented):
Code looks good! Have just commented on a couple of typos that need fixing 🙂

gbp committed:
Flatten application hash

Instead of properties hashes within application hash we now prepends
the hash keys so the values are accessible at `property_1_uprn`...

gbp committed:
Limit number of properties retrieved per application
